### PR TITLE
DNS Response Policy module

### PR DESCRIPTION
**Pull request by ludoo**
FAST modules will be updated to use response policies in a subsequent PR.

### CODE REVIEW

File: modules/dns-response-policy/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2022 Google LLC
```
Comment (LucaPrete) [inline modules/dns-response-policy/outputs.tf:2]: should we update to 2023?

File: modules/dns-response-policy/variables.tf [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2022 Google LLC
```
Comment (LucaPrete) [inline modules/dns-response-policy/variables.tf:2]: should we update to 2023?

File: modules/dns-response-policy/variables.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+variable "description" {
+  description = "Policy description."
+  type        = string
+  default     = "Terraform managed."
+}
+
+variable "name" {
+  description = "Zone name, must be unique within the project."
```
Comment (LucaPrete) [inline modules/dns-response-policy/variables.tf:31]: is this the policy name?

File: modules/dns-response-policy/versions.tf [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Google LLC
```
Comment (LucaPrete) [inline modules/dns-response-policy/versions.tf:1]: should we update to 2023?

File: modules/dns-response-policy/versions.tf [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 4.55.0" # tftest
+    }
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = ">= 4.55.0" # tftest
+    }
+    local = {
```
Comment (LucaPrete) [inline modules/dns-response-policy/versions.tf:26]: do we use this?

File: README.md [hunk 6]
```diff
@@ -30,7 +30,7 @@ The current list of modules supports most of the core foundational and networkin
...
+- **networking** - [DNS](./modules/dns), [DNS Response Policy](./modules/dns-response-policy/), [Cloud Endpoints](./modules/endpoints), [address reservation](./modules/net-address), [NAT](./modules/net-cloudnat), [Global Load Balancer (classic)](./modules/net-glb/), [L4 ILB](./modules/net-ilb), [L7 ILB](./modules/net-ilb-l7), [VPC](./modules/net-vpc), [VPC firewall](./modules/net-vpc-firewall), [VPC firewall policy](./modules/net-vpc-firewall-policy), [VPC peering](./modules/net-vpc-peering), [VPN dynamic](./modules/net-vpn-dynamic), [HA VPN](./modules/net-vpn-ha), [VPN static](./modules/net-vpn-static), [Service Directory](./modules/service-directory)
```
Comment (LucaPrete) [inline README.md:33]: should we start putting these in alphabetical order as well? Can also be another patch of course..

File: modules/dns-response-policy/README.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+# Google Cloud DNS Response Policy
+
+This module allows management of a [Google Cloud DNS policy and its rules](https://cloud.google.com/dns/docs/zones/manage-response-policies). The policy can be already existing and passed in via its id.
```
Comment (LucaPrete) [inline modules/dns-response-policy/README.md:3]: Feel free to ignore :)

The module allows managing a [Google Cloud DNS policy and its rules](https://cloud.google.com/dns/docs/zones/manage-response-policies). The policy can already exist and passed in via its id.

**Review by LucaPrete — APPROVED** [review]
very minor comments, otherwise lgtm

**Review by juliocc — APPROVED** [review]
I hope you don't mind that I added inventories to your two examples

File: modules/dns-response-policy/outputs.tf (hunk 1 above)
Comment (ludoo) [reply modules/dns-response-policy/outputs.tf:2]: yep good catch

File: modules/dns-response-policy/versions.tf (hunk 5 above)
Comment (ludoo) [reply modules/dns-response-policy/versions.tf:26]: we do, there was a nasty bug in the current version of locals; the versions file is unique anyway, so this identical to any other module in the repo :p

File: README.md (hunk 6 above)
Comment (ludoo) [reply README.md:33]: yep we should autogenerate this

**Comment by ludoo** [comment]
ouch, automerge kicked in before I could send the fix to copyright etc
